src/index.js

```diff
@@ -106,52 +120,49 @@ export function addMetadata({ types: t }) {
         }
       },
 
+      /**
+       * Do transforms for adding our test metadata expressions to beforeEach.
+       * For each top-level module (in QUnit there can be sibling and/or nested modules), only 1 beforeEach should apply,
+       * and so only 1 beforeEach will be added/transformed per top-level module. As Babel traverses through each call
+       * expression, we're only concerned about operating on call expressions inside of top-level module calls.
+       *
+       * While inside a top-level module, we check the current call expression for either of these conditions:
+       *   1. if it's a beforeEach, then we add our test metadata expressions to it.
+       *   2. if it's a call to either "test" or to a nested "module", then we want to add a new beforeEach just above it
+       * If the current call is neither of these, then do nothing. Babel will move on to the next call expression.
+       *
+       * The transformed beforeEach would look like:
+          hooks.beforeEach(function () {
+            let testMetadata = getTestMetadata(this);
+            testMetadata.filePath = 'test/my-test.js';
+          });
+       * @param {object} babelPath
+       * @param {object} state
+       */
       CallExpression(babelPath, state) {
-        // Reset at top-level module
-        if (babelPath.scope.block.type === 'Program')
-          beforeEachModified = false;
+        // If we're at a top-level call expression, then we reset the beforeEachModified state to false, and let Babel
+        // move on to the next call expression.
+        if (babelPath.scope.block.type === 'Program') {
+          state.opts.beforeEachModified = false;
+          return;
+        }
```

Comment:
Would it not be possible to just initialize this in the `Program` visitor?

Reply:
I just tried this out and unfortunately it doesn't work.

The case I am needing to reset for is whenever we are at a top-level module(). So that when we have sibling module calls like below, upon entering each `module`, `beforeEachModified` should be false.
```
module('Acceptance | browse acceptance test', function (hooks) {
  hooks.beforeEach(function () {
    let testMetadata = getTestMetadata(this);
    testMetadata.filePath =
      '__tests__/__fixtures__/multiple-module-no-beforeeach-test-skip-todo-code.js';
  });

  test('it renders browse page', async function (assert) {    ...  });
});

module('Acceptance | browse exclusive acceptance test', function (hooks) {
  hooks.beforeEach(function () {
    let testMetadata = getTestMetadata(this);
    testMetadata.filePath =
      '__tests__/__fixtures__/multiple-module-no-beforeeach-test-skip-todo-code.js';
  });

  test('it renders browse exclusive page', async function (assert) {    ...  });
});
```

Reply:
Hmmm we don't typically have sibling modules, but we do have nested ones. It's good for you to test and accommodate this though, since it's always possible someone has put multiple top-level module calls.